subsys/bluetooth/audio/vocs_client.c

```diff
@@ -0,0 +1,689 @@
+/*  Bluetooth VOCS - Volume Offset Control Service - Client */
+
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <zephyr/types.h>
+
+#include <device.h>
+#include <init.h>
+#include <sys/check.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/audio/vocs.h>
+
+#include "vocs_internal.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_VOCS_CLIENT)
+#define LOG_MODULE_NAME bt_vocs_client
+#include "common/log.h"
+
+static struct bt_vocs vocs_insts[CONFIG_BT_MAX_CONN * CONFIG_BT_VOCS_CLIENT_MAX_INSTANCE_COUNT];
+
+static struct bt_vocs *lookup_vocs_by_handle(struct bt_conn *conn, uint16_t handle)
+{
+	__ASSERT(handle != 0, "Handle cannot be 0");
+	__ASSERT(conn, "Conn cannot be NULL");
+
+	for (int i = 0; i < ARRAY_SIZE(vocs_insts); i++) {
+		if (vocs_insts[i].cli.conn == conn &&
+		    vocs_insts[i].cli.active &&
+		    vocs_insts[i].cli.start_handle <= handle &&
+		    vocs_insts[i].cli.end_handle >= handle) {
+			return &vocs_insts[i];
+		}
+	}
+
+	BT_DBG("Could not find VOCS instance with handle 0x%04x", handle);
+	return NULL;
+}
+
+uint8_t vocs_client_notify_handler(struct bt_conn *conn, struct bt_gatt_subscribe_params *params,
+				   const void *data, uint16_t length)
+{
+	uint16_t handle = params->value_handle;
+	struct bt_vocs *inst = lookup_vocs_by_handle(conn, handle);
+
+	if (!inst) {
+		BT_DBG("Instance not found");
+		return BT_GATT_ITER_STOP;
+	}
+
+	if (!data || !length) {
```

Comment:
Are you sure `length == 0` always means unsubscription, and that it isn't e.g. possible for the remote to send an empty payload which would result with `data != NULL && length == 0`?

Reply:
You are completely correct here :) Removed the BT_DBG, but I guess we can't do much else here.